src/Microsoft.Azure.WebJobs.Extensions.Redis/StreamTrigger/RedisStreamTriggerScaleMonitor.cs

```diff
@@ -11,21 +12,91 @@ internal class RedisStreamTriggerScaleMonitor : RedisPollingTriggerBaseScaleMoni
+            if (streamLength == 0)
             {
-                Remaining = multiplexer.GetDatabase().StreamLength(key),
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = 0,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            StreamGroupInfo[] groups = multiplexer.GetDatabase().StreamGroupInfo(key);
+            if (!groups.Any(g => g.Name == name))
+            {
+                // group doesn't exist, assume all entries in stream have not been processed
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = streamLength,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            StreamGroupInfo group = groups.First(g => g.Name == name);
+            if (multiplexer.GetServers()[0].Version >= RedisUtilities.Version70 && group.Lag.HasValue)
+            {
+                // Redis 7: Scaler gets number of remaining entries for the consumer group from XINFO GROUPS.
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = group.Lag.Value,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            StreamInfo info = await multiplexer.GetDatabase().StreamInfoAsync(key);
+            string firstId = (string)info.FirstEntry.Id;
+            string lastId = (string)info.LastEntry.Id;
+            string lastDeliveredId = group.LastDeliveredId ?? firstId;
+            if (lastId == lastDeliveredId)
+            {
+                // processed all entries in the stream
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = 0,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            // Redis 6/6.2 does not have an internal counter for remaining entries for the consumer group
+            // Estimate remaining entries using the time field from the entry ID (assuming ID="time-counter")
+            string[] firstIdSplit = firstId.Split('-');
+            string[] lastIdSplit = lastId.Split('-');
+            string[] lastDeliveredIdSplit = lastDeliveredId.Split('-');
+            long firstTimestamp = long.Parse(firstIdSplit[0]);
+            long lastTimestamp = long.Parse(lastIdSplit[0]);
+            long lastDeliveredTimestamp = long.Parse(lastDeliveredIdSplit[0]);
+
+            if (lastTimestamp == lastDeliveredTimestamp)
+            {
+                // If timestamp is the same, return counter difference
+                long lastCounter = long.Parse(lastIdSplit[1]);
+                long lastDelieveredCoutner = long.Parse(lastDeliveredIdSplit[1]);
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = Math.Min(streamLength, Math.Max(1, lastCounter - lastDelieveredCoutner)),
```

Comment:
Our of curious, what edge conditions we want to handle here

Reply:
This is specifically for the case if the customer was using custom IDs where they kept the `time` part of the ID as constant and only increased the `counter`. In this case, the estimation of remaining entries would be the difference between the counters of the last entry id and last entry delivered id. The counter can be any strictly increasing sequence, so I limit the `Remaining` to between 1 and the streamLength

Reply:
To learn, inaccurate count isn't impacting the correctness of result, even if in case it happens; rather just impact how many worker will be created?

Reply:
Yes, the value only impacts the number of requested workers and does not change any logic that processes messages from the stream.